### DISCUSSION TITLE
Upgrade after installing poetry downgrades dependencies

### DISCUSSION THREAD

**Discussion by BudzynskiMaciej** [python-poetry/poetry#9546] [Q&A] [answered]
Is it expected behavior that after installing poetry using an official script and then running the command: **_poetry self update_** this operation downgrades installed dependencies?

**Comment by radoering** [comment, marked as answer]
If you run `poetry self update` immediately after installing it, I would not expect any change. However, the script installs poetry via pip and `poetry self update` uses its own resolver so that it can happen that pip's resolver and poetry's resolver come up with different solutions. Apart from that, an update of some dependencies can always require a downgrade of other dependences.
